### PR TITLE
Initial implementation for timeouts via AbortController.

### PR DESCRIPTION
**Pull request by sblaurock**
Add ability to supply timeout via optional `networkParameters` for each method within each module.

### CODE REVIEW

File: spec/src/modules/catalog.js [hunk 1]
```diff
@@ -2230,7 +2446,7 @@ describe('ConstructorIO - Catalog', () => {
           fetch: fetchSpy,
         });
 
-        return expect(catalog.getRedirectRule({ id: redirectRuleId })).to.eventually.be.rejected;
+        return expect(catalog.getRedirectRules()).to.eventually.be.rejected;
```
Comment (sblaurock) [inline spec/src/modules/catalog.js:2449]: Fix previous tests - should be `getRedirectRules` not `getRedirectRule`

File: spec/src/modules/tracker.js [hunk 2]
```diff
@@ -2630,13 +2683,21 @@ describe('ConstructorIO - Tracker', () => {
     it('Should throw an error when invalid parameters are provided', () => {
       const { tracker } = new ConstructorIO({ apiKey: testApiKey });
 
-      expect(tracker.trackSearchResultClick(term, [], userParameters)).to.be.an('error');
+      expect(tracker.trackConversion(term, [], userParameters)).to.be.an('error');
```
Comment (jjl014) [inline spec/src/modules/tracker.js:2686]: nice catch! thanks for fixing the tests!

File: src/modules/autocomplete.js [hunk 3]
```diff
@@ -137,7 +142,12 @@ class Autocomplete {
       headers['User-Agent'] = userParameters.userAgent;
     }
 
-    return fetch(requestUrl, { headers }).then((response) => {
+    // Handle timeout if specified
+    if (networkParameters.timeout && typeof networkParameters.timeout === 'number') {
```
Comment (jjl014) [inline src/modules/autocomplete.js:146]: Should we do some destructuring here for `timeout`?

File: src/modules/browse.js [hunk 4]
```diff
@@ -213,7 +218,12 @@ class Browse {
       return Promise.reject(e);
     }
 
-    return fetch(requestUrl, { headers }).then((response) => {
+    // Handle timeout if specified
+    if (networkParameters.timeout && typeof networkParameters.timeout === 'number') {
```
Comment (jjl014) [inline src/modules/browse.js:222]: Some destructuring here as well

**Review by jjl014 — APPROVED** [review]
Nice work on this. Looks great to me!

Not a super big deal, but added some comments around destructuring. Was too lazy to add comments to all the files around destructuring the `timeout` from the `networkParameters`.

File: src/modules/autocomplete.js (hunk 3 above)
Comment (sblaurock) [reply src/modules/autocomplete.js:146]: We could - I don't really think it adds anything but extra lines of code?

Comment (jjl014) [reply src/modules/autocomplete.js:146]: I think it's fine to leave it as is for now until we add other options to the networkParameters? up to you 😄

Comment (sblaurock) [reply src/modules/autocomplete.js:146]: 👍  agreed! thanks for the review!

**Pull request merged**